src/ConstructiveSolidGeometry/ConstructiveSolidGeometry.jl

```diff
@@ -51,11 +51,16 @@ module ConstructiveSolidGeometry
     abstract type AbstractConstructiveGeometry{T} <: AbstractGeometry{T} end
     
     _csg_convert_args(eltype::Type{T}, r::Real) where T = convert(T, r) 
-    _csg_convert_args(eltype::Type{T}, r::Tuple) where T = (convert(T, r[1]), convert(T, r[2])) 
+    _csg_convert_args(eltype::Type{T}, r::Tuple{T1,T2}) where {T,T1,T2} = (_csg_convert_args(T, r[1]), _csg_convert_args(T, r[2]))
+    _csg_convert_args(eltype::Type{T}, r::Tuple{T1}) where {T,T1} = convert(T, r[1])
+    _csg_convert_args(eltype::Type{T}, r::Type{Nothing}) where T = nothing
+    _csg_convert_args(eltype::Type{T}, r::Nothing) where T = nothing
 
     _csg_get_promoted_eltype(::Type{T}) where {T <: AbstractArray} = eltype(T)
     _csg_get_promoted_eltype(::Type{T}) where {T <: Real} = T
-    _csg_get_promoted_eltype(::Type{Tuple{T1,T2}}) where {T1, T2} = promote_type(T1, T2)
+    _csg_get_promoted_eltype(::Type{Tuple{T1,T2}}) where {T1, T2} = promote_type.(T1, T2)
+    _csg_get_promoted_eltype(::Type{Tuple{T1,T2}},::Type{Tuple{T3,T4}}) where {T1, T2,T3,T4} = promote_type.(T1, T2,T3,T4)
```

Comment:
Here, I came across some issue for `c = CSG.Cone( r = ((0.4, 0.5), (0.4, 5)))`.
This should fix it:
```julia
_csg_get_promoted_eltype(::Type{Tuple{T1,T2}}) where {T1<:Real, T2<:Real} = promote_type(T1, T2)
_csg_get_promoted_eltype(::Type{Tuple{T1,T2}}) where {T1, T2} = promote_type(_csg_get_promoted_eltype(T1), _csg_get_promoted_eltype(T2))
```